### PR TITLE
[[ Bug 19909 ]] Add emscripten deploy library.

### PR DESCRIPTION
**Pull request by livecodemichael**
An internal command has been added to the Mac engine to return a
list of all the installed web browsers. We get a similar list on
Windows by querying the registry. For Linux we check a static
list of browsers to see what's installed.

The list of browsers is used as deploy targets for running HTML 5
standalones.

The deploy library creates an instance of a local HTTP server using
the HTTPd library, builds the standalone then launches the chosen
browser, pointing it to the appropriate localhost path.

Relies on https://github.com/livecode/livecode-ide/pull/1750

### CODE REVIEW

File: ide-support/revdeploylibraryemscripten.livecodescript [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+-- if it exists in the temp file space.
+--
+on deployEmscriptenNewHTTPRequest pSocketID, pRequest
+   local tPath   
+   put sHTTPFileSpace & slash & pRequest["resource"] into tPath
+   
+   if there is a file tPath then
+      httpdResponse pSocketID, 200, URL ("binfile:" & tPath)
+   else
+      httpdResponse pSocketID, 404
```
Comment (montegoulding) [inline ide-support/revdeploylibraryemscripten.livecodescript]: @livecodemichael do we want to display the emrun callbacks anywhere? I think they could just go to the msg box as when remote debugger is in use put is routed over the socket to the IDE to put into the msg box so it's a bit similar.

Comment (livecodemichael) [reply ide-support/revdeploylibraryemscripten.livecodescript]: @montegoulding Yeah, I was thinking about that. Seems like a good idea.

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 90fff0b

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 90fff0b39642023a2aec258c6d462be7cf305ea7

**Comment by livecode-vulcan** [comment]
😎 test success 90fff0b39642023a2aec258c6d462be7cf305ea7

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/842)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/858)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/832)
* try-community-universal-ios-iphoneos11.0: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.0/builds/7)
* try-community-universal-ios-iphonesimulator11.0: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.0/builds/8)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/836)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/858)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/842)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/858)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/841)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/851)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/873)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/837)

**Pull request merged**